community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/InternalAutoIndexOperations.java

```diff
@@ -0,0 +1,316 @@
+    public void propertyChanged( DataWriteOperations ops, long entityId, Property oldProperty, Property newProperty )
+            throws AutoIndexingKernelException
+    {
+        if ( enabled )
+        {
+            try
+            {
+                String name = propertyKeyLookup.getTokenById( oldProperty.propertyKeyId() ).name();
+                if ( propertyKeysToInclude.get().contains( name ) )
+                {
+                    ensureIndexExists( ops );
+                    type.remove( ops, entityId, name, oldProperty.value() );
+                    type.add( ops, entityId, name, newProperty.value() );
+                }
+            }
+            catch ( LegacyIndexNotFoundKernelException | EntityNotFoundException | PropertyNotFoundException e )
+            {
+                throw new AutoIndexingKernelException( e );
+            }
+            catch ( TokenNotFoundException e )
+            {
+                // TODO: TokenNotFoundException was added before there was a kernel. It should be converted to a
+                // KernelException now
+                throw new AutoIndexingKernelException( new PropertyKeyIdNotFoundKernelException( oldProperty
+                        .propertyKeyId(), e ) );
+            }
+        }
+    }
+
+    @Override
+    public void propertyRemoved( DataWriteOperations ops, long entityId, int propertyKey )
+            throws AutoIndexingKernelException
+    {
+        if ( enabled )
+        {
+            try
+            {
+                String name = propertyKeyLookup.getTokenById( propertyKey ).name();
+                if ( propertyKeysToInclude.get().contains( name ) )
+                {
+                    ensureIndexExists( ops );
+                    type.remove( ops, entityId );
+                }
+            }
+            catch ( LegacyIndexNotFoundKernelException | EntityNotFoundException e )
+            {
+                throw new AutoIndexingKernelException( e );
+            }
+            catch ( TokenNotFoundException e )
+            {
+                // TODO: TokenNotFoundException was added before there was a kernel. It should be converted to a
+                // KernelException now
+                throw new AutoIndexingKernelException( new PropertyKeyIdNotFoundKernelException( propertyKey, e ) );
+            }
+        }
+    }
+
+    @Override
+    public void entityRemoved( DataWriteOperations ops, long entityId ) throws AutoIndexingKernelException
+    {
+        if ( enabled )
+        {
+            try
+            {
+                ensureIndexExists( ops );
+                type.remove( ops, entityId );
+            }
+            catch ( LegacyIndexNotFoundKernelException | EntityNotFoundException e )
+            {
+                throw new AutoIndexingKernelException( e );
+            }
+        }
+    }
+
+    // Trap door needed to keep this as an enum
+    void replacePropertyKeysToInclude( List<String> propertyKeysToIncludeNow )
```

Comment:
as soon as propertyKeysToInclude will be included in constructor this method should go.